pkgs/build-support/ocaml/oasis.nix

```diff
@@ -0,0 +1,31 @@
+{ buildOcaml, ocaml_oasis }:
+
+{ name, version, buildInputs ? [], ...
+}@args:
+
+buildOcaml (args // {
```

Comment:
Please use `stdenv.mkDerivation` directly.

Reply:
But then I would duplicate the code in `buildOcaml`. Is that intended? If so could you quickly tell me the reasoning behind that?

Reply:
The `buildOcaml` helper function has been introduced for a specific set of libraries (the ones from Jane Street, at version approximately 113.33.xx). These legacy libraries are only available (in nixpkgs) for OCaml 4.02 and are bound to disappear.

Adding new packages that depend on this technical dept just creates a maintenance burden.